f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -832,21 +833,29 @@ def _post_init(self):
             self.network_builder.post_init()
 
         # read enhanced services definitions
+        self.init_esd()
+        self._set_agent_status(False)
+
+    def init_esd(self):
+        # read all esd file from esd dir
+        # init esd object in lbaas_builder
+        # init esd object in service_adapter
         esd_dir = os.path.join(self.get_config_dir(), 'esd')
-        esd = EsdTagProcessor(esd_dir)
+        # EsdTagProcessor is a singleton, so nothing new
```

Comment:
are you sure this is a singleton?

Reply:
sure

Reply:
see line #2412,  >1  init_esd() could lead here.